### ISSUE TITLE
pihole DNS not working

### ISSUE THREAD

**Issue by brent0np** [hassio-addons/addon-wireguard#13]
# Problem/Motivation

> I have followed the documentation to configure the addon, and like the [forum](https://community.home-assistant.io/t/community-hass-io-add-on-wireguard/134662) I am also having issues.  I have access to both and Android and iPhone I can use to test.

## Expected behaviour

> I expected to see the VPN client address (172.27.66.2 (android) or 172.27.66.3 (iPhone)) appear as a pihole client, or in the pihole network overview and for traffic to be filtered/reported in pihole.

## Actual behaviour

> When connected to the VPN using the android handset both the VPN client on the phone and the log in the hassio addon shows traffic sent/received for the relevant IP address.  If I access ipchicken.com from the android handset (while disconnected from my home wifi but connected to the VPN) it shows my home internet connections IP address correctly.  I also used the android app [pingtools](https://play.google.com/store/apps/details?id=ua.com.streamsoft.pingtools&hl=en_AU) while connected to the VPN.  It showed only one DNS server configured and a traceroute to google.com on the VPN worked using the expected route on the VPN, but still is not filtered by pihole.

With the iPhone, if I turn off wifi and then connect to the vpn I don't get any internet at all - which means I can't hit ipchicken.com.  I do see traffic stats in both the addon log and the VPN client on the iPhone and I can ping the ip address of the VPN specified DNS.

The bits I can't figure out;
- Why the android device can access the internet and the iPhone can't
- Why pihole doesn't recognise any of the VPN clients or the VPN network anywhere, but I can't successfully ping the ip address of the VPN specified DNS address.

## Steps to reproduce

> Install the addon following the documentation.  I have removed everything and reinstalled it twice to ensure I didn't miss anything.

## Proposed changes

> I suspect the VPN is working fine given traffic is going from client to concentrator(server) and back, but for some reason, the traffic is not getting into pihole.  Given the network doesn't appear to be known to pihole that would suggest there might be some configuration required on pihole?

**Comment by sinclairpaul** [comment]
Compare the configuration on the clients themself for your dns server, not what is set in the addon configuration (the addon config generates the conf/QR codes).  Pi-Hole will likely report the addon IP address as the source, as it is tunneled from there.

**Comment by phluther** [comment]
On the DNS page under settings for PiHole, have you set the option for PiHole to listen on all interfaces? Mine was set to listen only on eth0, and the connection worked using the DNS IP given in the readme only after changing this setting to all interfaces.

**Comment by harrywesterman** [comment]
> On the DNS page under settings for PiHole, have you set the option for PiHole to listen on all interfaces? Mine was set to listen only on eth0, and the connection worked using the DNS IP given in the readme only after changing this setting to all interfaces.

Bingo, that is it.

**Comment by brent0np** [comment]
Thank you all for providing some input. 

I now have it working.  I restarted the raspberry pi, recreated the device config so it generated another QR code(just to ensure it was all clean) and reconnected devices. (My pi is only connected using wifi)

Just for the sake of notes;
- The DNS on handset was always configured inline with the addon's settings and Traffic was hitting both ends
- **I changed pihole config to 'listen on all interfaces'**
- I can see the addon in the pihole client list as suggested

I expect that changing that setting was the issue.  Thanks for the advice all.
